pkg/server/datastore/sqldriver/awsrds/auth_token.go

```diff
@@ -0,0 +1,103 @@
+package awsrds
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+)
+
+type authTokenBuilder interface {
+	buildAuthToken(ctx context.Context, endpoint string, region string, dbUser string, creds aws.CredentialsProvider, optFns ...func(options *auth.BuildAuthTokenOptions)) (string, error)
+}
+
+type tokenGetter interface {
+	getAWSAuthToken(ctx context.Context, params *Config, tokenBuilder authTokenBuilder) (string, error)
```

Comment:
I'd rename to `getAuthToken` to keep it consistent with the naming in the `authTokenBuilder` interface.